packages/opentelemetry-instrumentation-mcp/opentelemetry/instrumentation/mcp/instrumentation.py

```diff
@@ -0,0 +1,134 @@
+from typing import Collection
+import mcp
+import json
+import asyncio
+import nest_asyncio
+nest_asyncio.apply()
+
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.trace import get_tracer
+from wrapt import wrap_function_wrapper as _W
+from opentelemetry.trace.status import Status, StatusCode
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+from opentelemetry.trace.propagation import set_span_in_context
+from opentelemetry.semconv_ai import SpanAttributes
+
+from opentelemetry.instrumentation.mcp.version import __version__
+
+_instruments = ("mcp >= 1.3.0",)
+
+class McpInstrumentor(BaseInstrumentor):
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+    
+    def _instrument(self, **kwargs):
+        tracer_provider = kwargs.get("tracer_provider")
+        tracer = get_tracer(__name__, __version__, tracer_provider)
+
+        _W(
+            "mcp.server.lowlevel.server",
+            "Server._handle_request",
```

Comment:
a bit smelly and prone to be break the fact we're wrapping "private" method. is there any "public" wrapper for it?